index.d.ts

```diff
@@ -0,0 +1,70 @@
+import { ComponentClass, Component, StatelessComponent, ReactNode } from 'react';
+import { Store, Dispatch, ActionCreator } from 'redux';
+
+interface ComponentDecorator<TOriginalProps, TOwnProps> {
+  (component: ComponentClass<TOriginalProps>|StatelessComponent<TOriginalProps>): ComponentClass<TOwnProps>;
+}
+
+/**
+ * Following 3 functions cover all possible ways connect could be invoked
+ *
+ * - State: Redux state interface (the same one used by Store<S>)
+ * - TStateProps: Result of MapStateToProps
+ * - TDispatchProps: Result of MapDispatchToProps
+ * - TOwnProps: Props passed to the wrapping component
+ */
+export function connect<State, TOwnProps>(): ComponentDecorator<{ dispatch: Dispatch<State> }, TOwnProps>;
+
+export function connect<State, TOwnProps, TStateProps>(
+  mapStateToProps: MapStateToProps<State, TStateProps, TOwnProps>,
+): ComponentDecorator<TStateProps & { dispatch: Dispatch<State> }, TOwnProps>;
+
+export function connect<State, TOwnProps, TStateProps, TDispatchProps>(
+  mapStateToProps: MapStateToProps<State, TStateProps, TOwnProps>|null,
+  mapDispatchToProps: MapDispatchToPropsFunction<State, TDispatchProps, TOwnProps>|TDispatchProps
+): ComponentDecorator<TStateProps & TDispatchProps, TOwnProps>;
+
+export function connect<State, TOwnProps, TStateProps, TDispatchProps>(
+  mapStateToProps: MapStateToProps<State, TStateProps, TOwnProps>|null,
+  mapDispatchToProps: MapDispatchToPropsFunction<State, TDispatchProps, TOwnProps>|TDispatchProps,
+  mergeProps: MergeProps<TStateProps, TDispatchProps, TOwnProps>,
+  options?: Options
+): ComponentDecorator<TStateProps & TDispatchProps, TOwnProps>;
+
+interface MapStateToProps<State, TStateProps, TOwnProps> {
+  (state: State, ownProps: TOwnProps): TStateProps;
+}
+
+/**
+ * State is not actually used here but included for consistency with Redux typings and MapStateToProps.
+ */
+interface MapDispatchToPropsFunction<State, TDispatchProps, TOwnProps> {
+  (dispatch: Dispatch<State>, ownProps: TOwnProps): TDispatchProps;
+}
+
+/**
+ * Any since not every ActionCreator returns the same Action
+ */
+interface MapDispatchToPropsObject<TDispatchProps> {
+  [name: string]: ActionCreator<TDispatchProps>;
+}
+
+interface MergeProps<TStateProps, TDispatchProps, TOwnProps> {
+  (stateProps: TStateProps, dispatchProps: TDispatchProps, ownProps: TOwnProps): TStateProps & TDispatchProps;
+}
+
+interface Options {
+  pure?: boolean;
+  withRef?: boolean;
+}
+
+/**
+ * Typescript does not support generic components in tsx yet in an intuïtive way which is the reason we avoid a
+ * generic parameter in Store here by using any as the type
+ */
+export interface ProviderProps {
+  store?: Store<any>;
+  children?: ReactNode;
```

Comment:
That's not needed as `children` prop is automatically added to components defined as classes.